### PR TITLE
fix(types): incorrect return type of collectDefaultMetrics(), should be Timeout in Node

### PR DESCRIPTION
**Pull request by nazar-pc**

### CODE REVIEW

**Comment by SimenB** [comment]
Thanks! Mind adding a `/// <reference types="node" />` at the top of the file?

**Comment by nazar-pc** [comment]
Looks like `@types/node` dependency is needed. Ideally in `dependencies`.

**Comment by SimenB** [comment]
Let's do dev dependencies so we don't pollute consumers' code with it. Ref https://github.com/facebook/jest/issues/8092

**Comment by siimon** [comment]
Can you please look into the broken ci?

**Comment by nazar-pc** [comment]
Updated dependencies, should be good now

File: index.d.ts [hunk 1]
```diff
@@ -655,7 +655,7 @@ export interface DefaultMetricsCollectorConfiguration {
  */
 export function collectDefaultMetrics(
 	config?: DefaultMetricsCollectorConfiguration
-): number;
+): NodeJS.Timeout;
```
Comment (SimenB) [inline index.d.ts]: Do we not the the references comment at the top?

Comment (nazar-pc) [reply index.d.ts]: Looks like it is not needed?

Comment (nazar-pc) [reply index.d.ts]: Looks like it is not needed? Actually, it is possible to use something like `ReturnType` to not even specify it explicitly if you want that.

Comment (SimenB) [reply index.d.ts]: `ReturnType<typeof global.setTimeout`? Might work, although since this package only works in Node it should be fine

Comment (SimenB) [reply index.d.ts]: If the references part is not needed, great!

Comment (nazar-pc) [reply index.d.ts]: Refactored to that

File: index.d.ts [hunk 2]
```diff
@@ -655,7 +655,7 @@ export interface DefaultMetricsCollectorConfiguration {
  */
 export function collectDefaultMetrics(
 	config?: DefaultMetricsCollectorConfiguration
-): number;
+): ReturnType<typeof setTimeout>;
```
Comment (SimenB) [inline index.d.ts]:
```suggestion
): ReturnType<typeof setInterval>;
```

**Comment by SimenB** [comment]
If you remove the unneeded package(-lock).json changes (now that we use the return type), we can merge this 🙂

**Comment by nazar-pc** [comment]
I've changed as suggested, tests are good. Globals are strange, but I've tested in my project and it seems to work as it should.

**Comment by SimenB** [comment]
https://github.com/siimon/prom-client/releases/tag/v11.5.0

**Pull request merged**